src/common/plugins/plugins-config.js

```diff
@@ -88,41 +92,82 @@ const getModel = (options: KPOptionsObject): Object => {
 };
 
 /**
- * @param {KPOptionsObject} options - player options
+ * @param {PKPluginsConfigObject} options - plugins options
+ * @param {KPOptionsObject} config - player config
  * @private
  * @return {void}
  */
-function evaluatePluginsConfig(options: KPOptionsObject): void {
-  if (options.plugins) {
-    pluginConfig.set(options.plugins);
-    const dataModel = getModel(options);
-    const evaluatedConfig = evaluate(JSON.stringify(pluginConfig.get()), dataModel);
-    let evaluatedConfigObj;
-    try {
-      evaluatedConfigObj = JSON.parse(evaluatedConfig, function(key) {
-        try {
-          return JSON.parse(this[key]);
-        } catch (e) {
-          return this[key];
-        }
-      });
-    } catch (e) {
-      evaluatedConfigObj = {};
-    }
-    evaluatedConfigObj = removeUnevaluatedExpression(evaluatedConfigObj);
-    options.plugins = removeUnevaluatedExpression(options.plugins);
+function evaluatePluginsConfig(options: PKPluginsConfigObject, config: KPOptionsObject): void {
+  if (options) {
+    pluginConfig.set(options);
+    const dataModel = getModel(config);
+    const mergedConfig = Utils.Object.mergeDeep({}, pluginConfig.get(), options);
+    const evaluatedConfig = _formatConfigString(evaluate(JSON.stringify(mergedConfig), dataModel));
+    _mergeConfig(options, evaluatedConfig);
+  }
+}
 
-    if (options.plugins) {
-      Object.keys(options.plugins).forEach(pluginName => {
-        if (options.plugins && options.plugins[pluginName]) {
-          const mergedConfig = Utils.Object.mergeDeep({}, evaluatedConfigObj[pluginName], options.plugins[pluginName]);
-          if (options.plugins) {
-            options.plugins[pluginName] = mergedConfig;
-          }
+/**
+ * @param {UIOptionsObject} options - UI options
+ * @param {KPOptionsObject} config - player config
+ * @private
+ * @return {void}
+ */
+function evaluateUIConfig(options: UIOptionsObject, config: KPOptionsObject): void {
+  if (options) {
+    const defaultUiConfig = {
```

Comment:
consider move to store (like plugins)

Reply:
I agree, but we currently have a bug with current implementation, which is shared across player instances so didn't want to recreate it here.
Once we move to new token evaluation we can create one big happy store and fix the static dependency.